### PR TITLE
Refactor all the commands to be Commands from the Args package.  Also use CommandRunner for the top-level command.

### PR DESCRIPTION
**Pull request by iansf**
This will break travis until I can get something like https://github.com/dart-lang/args/pull/40 pushed to a new version of the args package (and I update the pubspec.yaml). I guess I should hold off on pushing until then?

### CODE REVIEW

File: bin/sky_tools.dart [hunk 1]
```diff
@@ -2,185 +2,147 @@
...
+    }
+    if (rec.stackTrace != null) {
+      print(rec.stackTrace);
+    }
   });
+
+  var runner =
+      new FlutterCommandRunner();
+
+  runner
```
Comment (abarth) [inline bin/sky_tools.dart]: No need for this line.  If you drop the ; on the previous line, it will work.

**Comment by abarth** [comment]
LGTM

**Comment by iansf** [comment]
Thanks!  Do you think I should wait for the args package update to push this, or just tolerate the broken tests on travis?  (The tests pass locally since I have a dependency override for args to my version.)

**Comment by abarth** [comment]
I think we should wait until the args package publishes.  Otherwise sky_tools will be in an unreleasable state.

**Comment by jamesr** [comment]
I'd like to keep sky_tools releasable for now as I'm pushing fixes for the run_mojo workflow.

**Comment by iansf** [comment]
The args package team appears to prefer a different trade-off between mockability and private interfaces than I do, so I've rewritten the tests in this version.  Anyone care to take a look again?  Only init_test.dart, install_test.dart, and common.dart are different.

**Comment by abarth** [comment]
LGTM

**Pull request merged**